sphinx/ext/cylc_lang.py

```diff
@@ -118,24 +127,37 @@ class CylcLexer(RegexLexer):
         'multiline-setting': [
             (r'\"\"\"', String.Double, '#pop'),
             include('comment'),
-            include('jinja2-openers'),
+            include('preproc'),
             (r'(\n|.)', String.Double)
         ],
 
         # Graph strings:  foo => bar & baz
         'graph': [
-            include('jinja2-openers'),
+            include('preproc'),
             include('comment'),
+            include('inter-suite-trigger'),
             include('parameterisation'),
+            (r'@\w+', GRAPH_XTRIGGER_TOKEN),
             (r'\w+', GRAPH_TASK_TOKEN),
+            (r'\!\w+', Other),
             (r'\s', Text),
             (r'=>', Operator),
-            (r'[\&\|\!]', Operator),
+            (r'[\&\|]', Operator),
             (r'[\(\)]', Punctuation),
             (r'\[', Text, 'intercycle-offset'),
```

Comment:
I have noticed one omission. Dependencies on specific cycle points e.g. ``graph = "red[2019] => blue"`` are supported as valid specification within square brackets in a graph string, where only intercycle offsets appear to be recognised by the lexer as-is.

Reply:
The good news is the lexer is now so water-tight it picked up on several missing markers in some of the Cylc documentation code excerpts. It's nearly there :clap: